sprint-mission-6.sql

```diff
@@ -0,0 +1,220 @@
+        --     four_cheese_l | four_cheese | L    | 17.95 |           1316
+        --     classic_dlx_m | classic_dlx | M    |    16 |           1181
+        --     spicy_ital_l  | spicy_ital  | L    | 20.75 |           1109
+        --     hawaiian_s    | hawaiian    | S    |  10.5 |           1020
+        --     southw_ckn_l  | southw_ckn  | L    | 20.75 |           1016
+        --     bbq_ckn_l     | bbq_ckn     | L    | 20.75 |            992
+        --     bbq_ckn_m     | bbq_ckn     | M    | 16.75 |            956
+        -- ```
+
+
+
+
+    -- 2. `orders` 테이블에서 2025년 3월의 일별 주문 수량을 `total_orders`라는 이름으로, 일별 총 주문 금액을 `total_amount`라는 이름으로 포함해서 조회하세요.
+        select o.date, count(distinct o.id) as "total_orders", sum(od.quantity * p.price) as "total_amount"
+        from orders as o
+        join order_details as od
+        on od.order_id = o.id
+        join pizzas as p
+        on od.pizza_id = p.id
+        where o.date >= '2025-03-01' and o.date < '2025-04-01'
+        group by o.date
+        order by o.date;
+    --     출력 예시:
+        
+    --     ```sql
+    --     2025-03-01 |           49 | 1598.5500011444092
+    --     2025-03-02 |           58 |  2379.050001144409
+    --     2025-03-03 |           53 | 2287.8999996185303
+    --     2025-03-04 |           59 |  2444.300001144409
+    --     2025-03-05 |           64 |  2350.650005340576
+    --     ```
+
+    
+
+-- 
+--     3. `order`의 `id`가 78에 해당하는 주문 내역들을 조회합니다. 주문 내역에서 각각 주문한 피자의 이름을 `pizza_name`, 피자의 크기를 `pizza_size`, 피자 가격을 `pizza_price`, 수량을 `quantity`, 각 주문 내역의 총 금액을 `total_amount` 라는 이름으로 조회해 주세요.
+        select pt.name as "pizza_name", p.size as "pizza_size", p.price as "pizza_price", od.quantity as "quantity", (od.quantity * p.price) as "total_amount"
+        from orders as o
+        join order_details as od
+        on o.id = od.order_id
+        join pizzas as p 
+        on od.pizza_id = p.id
+        join pizza_types as pt
+        on p.type_id = pt.id
+        where o.id = 78;
+
+--         출력 예시:
+        
+--         ```sql
+--         The Thai Chicken Pizza      | S          |       12.75 |        1 |              12.75
+--         The Big Meat Pizza          | S          |          12 |        1 |                 12
+--         The Classic Deluxe Pizza    | S          |          12 |        1 |                 12
+--         The Italian Capocollo Pizza | M          |          16 |        1 |                 16
+--         The Spicy Italian Pizza     | L          |       20.75 |        3 |              62.25
+--         The Four Cheese Pizza       | L          |       17.95 |        1 | 17.950000762939453
+--         ```
+-- 
+
+
+
+    -- 4. `order_details`와 `pizzas` 테이블을 JOIN해서 피자 크기별(S, M, L) 총 수익을 계산하고, 크기별 수익을 출력하세요.
+        select p.size, sum(od.quantity * p.price)
+        from order_details as od
+        join pizzas as p
+        on od.pizza_id = p.id
+        group by p.size;
```

Comment:
제공된 아래 출력 예시에 따라, 약간 수정해본다면 아래처럼 정렬할 수 있겠네요!

```suggestion
SELECT p.size, SUM(od.quantity * p.price) AS total_revenue
FROM order_details od
JOIN pizzas p ON od.pizza_id = p.id
GROUP BY p.size
ORDER BY total_revenue DESC;
```